packages/lab/src/list-next/useList.ts

```diff
@@ -84,6 +103,13 @@ export const useList = ({
     [setSelectedItem, updateActiveDescendant]
   );
 
+  // Effect to move the cursor when items change controlled.
+  // this could be following active descendant if there is no better way of doing it when controlled
+  useEffect(() => {
+    const activeOptions = getOptions();
+    highlightedIndex && updateScroll(activeOptions[highlightedIndex]);
+  }, [highlightedIndex, getOptions, updateScroll]);
```

Comment:
is there a way to directly drive `activeDescendant` via `useControlled`'s `controlled` prop?

Reply:
there isn't, but active descendant always follows highlightedIndex (changed to to highlighteditem to move them all with value)